### PR TITLE
PD-651 Capture user email to add to Segment page properties

### PR DESCRIPTION
**Pull request by edwardupton**

### CODE REVIEW

File: src/segmentTracker/helpers/addEmailToEvents.ts [hunk 1]
```diff
@@ -15,10 +16,19 @@ export const addEmailToTrackEvents = ({ payload, next }: MiddlewareOptions): voi
...
-	if (email) {
-		properties.email = email;
+	if (emailFromTraits) {
+		// user() is not available on page load, so need to cache for next page event
+		window.localStorage.setItem(localStorageNames.email, emailFromTraits);
+		properties.email = emailFromTraits;
+
+		return properties;
+	}
+	const emailFromStorage = window.localStorage.getItem(localStorageNames.email);
```
Comment (lyykfi) [inline src/segmentTracker/helpers/addEmailToEvents.ts:28]: newline after if please.

File: src/segmentTracker/helpers/addEmailToEvents.ts [hunk 2]
```diff
@@ -15,10 +16,19 @@ export const addEmailToTrackEvents = ({ payload, next }: MiddlewareOptions): voi
 };
 
 const addEmailToProperties = (properties: LooseObject): LooseObject => {
-	const email = window.analytics.user && window.analytics.user().traits().email;
+	const emailFromTraits = window.analytics?.user?.().traits().email;
 
-	if (email) {
-		properties.email = email;
+	if (emailFromTraits) {
```
Comment (lyykfi) [inline src/segmentTracker/helpers/addEmailToEvents.ts:21]: should we have tests for that?

File: src/segmentTracker/helpers/addEmailToEvents.ts [hunk 3]
```diff
@@ -15,10 +16,19 @@ export const addEmailToTrackEvents = ({ payload, next }: MiddlewareOptions): voi
...
+		// user() is not available on page load, so need to cache for next page event
+		window.localStorage.setItem(localStorageNames.email, emailFromTraits);
```
Comment (lyykfi) [inline src/segmentTracker/helpers/addEmailToEvents.ts:23]: I think we have a copy/paste here.
let try to keep it a bit simpler.

`
if (emailFromTraits) {
 const emailFromStorage = window.localStorage.getItem(localStorageNames.email);

if (!emailFromStorage) {
 window.localStorage.setItem(localStorageNames.email, emailFromTraits);
}

properties.email = emailFromStorage  ?? emailFromTraits;
}

return properties;
`

wdyt?

File: src/segmentTracker/helpers/addEmailToEvents.ts (hunk 2 above)
Comment (edwardupton) [reply src/segmentTracker/helpers/addEmailToEvents.ts:21]: added tests

Comment (lyykfi) [reply src/segmentTracker/helpers/addEmailToEvents.ts:21]: ❤️

**Review by lyykfi — APPROVED** [review]

**Pull request merged**